src/utils.ts

```diff
@@ -124,6 +124,13 @@ function interopDefault(mod: any): any {
         return true;
       }
       if (prop === "default") {
+        if (
+          !defIsNil &&
+          typeof def === "object" &&
+          typeof def.default === "function"
```

Comment:
(sorry for late reply) wondering, shall we scope down the fix by also checking `__esModule` existence? (i'm mostly worried about behavior regressions)